src/ControlSystem/ApparentHorizons/Measurements.hpp

```diff
@@ -0,0 +1,125 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+
+#include "ApparentHorizons/ComputeHorizonVolumeQuantities.hpp"
+#include "ApparentHorizons/Tags.hpp"
+#include "ControlSystem/Protocols/Measurement.hpp"
+#include "ControlSystem/Protocols/Submeasurement.hpp"
+#include "ControlSystem/RunCallbacks.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/LinkedMessageQueue.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "ParallelAlgorithms/Interpolation/Callbacks/ErrorOnFailedApparentHorizon.hpp"
+#include "ParallelAlgorithms/Interpolation/Callbacks/FindApparentHorizon.hpp"
+#include "ParallelAlgorithms/Interpolation/Interpolate.hpp"
+#include "ParallelAlgorithms/Interpolation/Targets/ApparentHorizon.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t VolumeDim>
+class ElementId;
+template <size_t Dim>
+class Mesh;
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+namespace domain::Tags {
+template <size_t Dim>
+struct Mesh;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace control_system::ah {
+enum class HorizonLabel { AhA, AhB };
+std::string name(const HorizonLabel h) {
+  return h == HorizonLabel::AhA ? "AhA" : "AhB";
+}
+
+struct BothHorizons : tt::ConformsTo<protocols::Measurement> {
+  template <HorizonLabel Horizon>
+  struct FindHorizon : tt::ConformsTo<protocols::Submeasurement> {
+   private:
+    template <typename ControlSystems>
+    struct Interpolator {
+      static std::string name() {
+        return "ControlSystem::BothHorizons::" + ah::name(Horizon);
+      }
+
+      struct temporal_id : db::SimpleTag {
+        using type = LinkedMessageId<double>;
+      };
+
+      using vars_to_interpolate_to_target =
+          tmpl::list<gr::Tags::SpatialMetric<3, ::Frame::Grid, DataVector>,
```

Comment:
Don't we find the horizon in the distorted frame? Or is support for that not completed yet?

Reply:
I copied these lists from the ones in the BBH metavariables.  I do not know if those are out-of-date.

Reply:
Ok we should probably keep them the same for now so I'd say just leave it. Probably a question @markscheel can answer.

Reply:
we will ultimately need to find the AHs in the distorted frame, but I think that should be done later. It'll need more maps, etc. in the DataBox. It's not a small change.

Reply:
Yes, let's add distorted frame later. There are a few changes in the interpolation framework that need to come first.